File: Semestr 6/JavaProjekt/Projekt/src/main/java/com/example/projekt/models/Employee.java
package com.example.projekt.models;

import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Employee {
    @Id
    @GeneratedValue
    private int id;

    @OneToOne
    private User user;

    private String name;
    private String lastName;
    private String email;
    private String JobTitle;

    @OneToOne(targetEntity = Factory.class)
    @JoinColumn(referencedColumnName = "id")
    private Factory factory;

    @OneToOne(targetEntity = Department.class)
    @JoinColumn(referencedColumnName = "id")
    private Department department;

    @OneToOne(targetEntity = Computer.class)
    @JoinColumn(referencedColumnName = "id")
    private Computer computer;
}

File: Semestr 6/JavaProjekt/Frotnend/src/app/Models/Employee.ts
import { Computer } from "./Computer";
import { Department } from "./Department";
import { Factory } from "./Factory";

export interface Employee {
    id: number;
    name: string;
    lastName: string;
    email: string;
    jobTitle: string;
    factory: Factory;
    department: Department;
    computer: Computer
}

export interface EmployeePC {
    id: number;
    name: string;
    lastName: string;
}